app/controllers/admin/edition_images_controller.rb

```diff
@@ -26,7 +26,11 @@ def create
     @new_image = @edition.images.build
     @new_image.build_image_data(image_params["image_data"])
 
-    if @new_image.save
+    unless @new_image.filename_is_unique?
+      @new_image.image_data.errors.add(:file, message: "called #{@new_image.image_data.file.file&.filename} has already been uploaded. Choose a different image or rename the file.")
+    end
```

Comment:
This might be nicer if implemented in the model, rather than directly in the controller. It feels more like a 'data layer' concern than something the controller should be aware of.

It should be possible (I think?) to move this into a [custom validation method](https://guides.rubyonrails.org/active_record_validations.html#custom-methods) on the model.